src/controller/task.spec.ts

```diff
@@ -0,0 +1,215 @@
+    before('Prepare test data', async () => {
+      await TaskModel.create(Tasks)
+    })
+    after('Delete test data', async () => {
+      await TaskModel.deleteMany()
+    })
+    it('should get all tasks', async () => {
+      const response = await axios.get('/tasks')
+      expect(response.status).to.equal(200)
+      expect(response.data).to.be.a('object')
+    })
+  })
+  describe('GET by Id', () => {
+    before('Prepare test data', async () => {
+      await TaskModel.create(Tasks)
+    })
+    after('Delete test data', async () => {
+      await TaskModel.deleteMany()
+    })
+    it('should get tasks by id', async () => {
+      const response = await axios.get(`/tasks/${validId}`)
+      expect(response.status).to.equal(200)
+      expect(response.data).to.be.a('object')
+    })
+    it('should return an error 400 for Invalid Id', async () => {
+      try {
+        await axios.get(`/tasks/${invalidId}`)
+        expect.fail('call should have failed')
+      } catch (err: any) {
+        const { response: { status, data } } = err
+        expect(status).to.equal(400)
+        expect(data.message).to.equal('Validation error')
+      }
+    })
+    it('should return an error 404 for non-existent Id', async () => {
+      try {
+        await axios.get(`/tasks/${nonExisted}`)
+        expect.fail('call should have failed')
+      } catch (err: any) {
+        const { response: { status, data } } = err
+        expect(status).to.equal(404)
+        expect(data.message).to.equal('Task not found')
+      }
+    })
+  })
+  describe('DELETE by Id', () => {
+    before('Prepare test data for', async () => {
+      await TaskModel.create(Tasks)
+    })
+    after('Delete test data', async () => {
+      await TaskModel.deleteMany()
+    })
+    it('should delete tasks by id', async () => {
+      const response = await axios.delete(`/tasks/${validId}`)
+      expect(response.status).to.equal(204)
+    })
+    it('should return an error 404 for non-existent Id', async () => {
+      try {
+        await axios.delete(`/tasks/${nonExisted}`)
+        expect.fail('call should have failed')
+      } catch (err: any) {
+        const { response: { status, data } } = err
+        expect(status).to.equal(404)
+        expect(data.message).to.equal('Task not found')
+      }
+    })
+    it('should return an error 400 for invalid Id', async () => {
+      try {
+        await axios.delete(`/tasks/${invalidId}`)
+        expect.fail('call should have failed')
+      } catch (err: any) {
+        const { response: { status, data } } = err
+        expect(status).to.equal(400)
+        expect(data.message).to.equal('Validation error')
+      }
+    })
+  })
+  describe('PUT', () => {
+    before('Prepare test data for', async () => {
+      await TaskModel.create(Tasks)
+    })
+    after('Delete test data', async () => {
+      await TaskModel.deleteMany()
+    })
+    it('should replace a task', async () => {
+      const UpdateTask = {
```

Comment:
Why this is `UpdateTask` instead of `updateTask` in case we follow camelCase style ?